npm/vite-dev-server/src/getVite.ts

```diff
@@ -1,21 +1,92 @@
+import module from 'module'
+import path from 'path'
+import os from 'os'
+import type { ViteDevServerConfig } from './devServer.js'
+import majorVersion from 'semver/functions/major.js'
 
 const debug = debugFn('cypress:vite-dev-server:getVite')
 
-export type Vite = typeof import('vite-6')
+export type Vite = typeof import('vite-7')
+
+class CJSNotSupportedError extends Error {
+  constructor (message: string) {
+    super(message)
+    this.name = 'CJSNotSupportedError'
+  }
+}
 
 // "vite-dev-server" is bundled in the binary, so we need to require.resolve "vite"
 // from root of the active project since we don't bundle vite internally but rather
 // use the version the user has installed
-export function getVite (config: ViteDevServerConfig): Vite {
+export async function getVite (config: ViteDevServerConfig): Promise<Vite> {
+  const filePrefix = os.platform() === 'win32' ? 'file://' : ''
+
   try {
-    const viteImportPath = require.resolve('vite', { paths: [config.cypressConfig.projectRoot] })
+    const require = module.createRequire(import.meta.url)
+    const vitePackageJsonPath = require.resolve('vite/package.json', { paths: [config.cypressConfig.projectRoot] })
+    const vitePackageJson = (await import(`${filePrefix}${vitePackageJsonPath}`, {
+      with: {
+        type: 'json',
+      },
+    })).default
+
+    const viteExports = vitePackageJson.exports['.']
+    let esmPath = null
+    let cjsPath = null
+
+    // In Node 20, require.resolve in the ESM context returns the CJS path as if we were in a CJS context.
+    // In Node 22, this is not the case and the ESM context is returned correctly.
+    // In order to work around this, we need to check where the ESM path is so we can import the correct path.
+    // In Vite 7, the CJS build was removed so there is only a single string entry in the export.
+    // Otherwise, both builds exists in Vite 6 and under and we only want to get the ESM path.
+
+    const majorVersionNumber = majorVersion(vitePackageJson.version)
+
+    if (majorVersionNumber >= 7) {
+      esmPath = viteExports
+    } else if (majorVersionNumber === 5) {
+      esmPath = viteExports.import.default
+      cjsPath = viteExports.require.default
+    } else {
+      esmPath = viteExports.import
+      cjsPath = viteExports.require
+    }
 
-    debug('resolved viteImportPath as %s', viteImportPath)
+    debug('vite ESM build path: %s', esmPath)
+    debug('vite CJS build path: %s', cjsPath)
 
-    return require(viteImportPath)
+    try {
+      // try to import the ESM build of Vite
+      const esmViteImportPath = path.resolve(vitePackageJsonPath, '../', esmPath)
+
+      debug('resolved esmViteImportPath as %s', esmViteImportPath)
+
+      const viteImport = await import(`${filePrefix}${esmViteImportPath}`)
+
+      return viteImport
+    } catch (err) {
+      if (majorVersionNumber >= 7) {
+        throw new CJSNotSupportedError(`CJS builds of vite ${majorVersionNumber} are not supported`)
```

Comment:
@AtofStryker Is there some action we can add to the error message here for the user to take to resolve this issue?